### ISSUE TITLE
invariant_booleans false positive with unary operations.

### ISSUE THREAD

**Issue by dikmax** [dart-lang/linter#337]
Consider this code:

```dart
if (offset >= length) {
  return;
}

offset++;
if (offset >= length) { // Got error here.
}
```

With `+=` works fine.

Linter 0.1.29

**Comment by dikmax** [comment]
As mentioned #334, I can be sure that it reproducible in 0.1.27 (version bundled with Dart 1.20.1), not in 0.1.29.

**Comment by alexeieleusis** [comment]
This is indeed a bug, those operators were not considered when looking for mutations, this rule needs some cleanup, I will fix this bug after that. It will take a while, but will be fixed.
